applications/nrf_desktop/README.rst

```diff
@@ -1652,6 +1652,64 @@ For example, the following line will start the upload of the new image to the de
 
   mcumgr -t 60 --conntype serial --connstring=/dev/ttyACM0 image upload build-nrf52833dongle_nrf52833/zephyr/app_update.bin
 
+The fwupd support
+=================
+
+The fwupd is an open-source project providing tools and deamon for managing the installation of firmware updates on Linux-based systems.
+Together with the Linux Vendor Firmware Service (LVFS) it provide a solution for ventors to easily distribute firmware for compatible devices.
+
+The fwupd tools can communicate with the devices running nRF Desktop application with :ref:`nrf_desktop_bootloader_background_dfu` feature enabled.
+
+Nordic HID plugin
+-----------------
+
+The fwupd project allows communication with multiple types of devices through various communication protocols.
+The communication protocols are implemented using plugins.
+The plugin associated with the DFU protocol realized through the :ref:`nrf_desktop_config_channel` is branded as ``nordic_hid``.
+
+Adding a new device
+-------------------
+
+The device specifies which protocol is used for the communication by adding the specific device information to the plugin metadata.
+For the ``nordic_hid`` plugin this file is located at ``https://github.com/fwupd/fwupd/tree/main/plugins/nordic-hid/nordic-hid.quirk``
+The example entry is presented below:
+
+.. code-block:: console
+
+   [HIDRAW\VEN_1915&DEV_52DE]
+   Plugin = nordic_hid
+   GType = FuNordicHidCfgChannel
+   NordicHidBootloader = B0
+
+The first line describes the method of communication (``HIDRAW`` selects communication using ``HID`` reports) as well as vendor and device identifiers.
+This tells the fwupd tool where it should look for the device and how it should be identified.
+The next two lines set the plugin being used by the device.
+The final line ``NordicHidBootloader`` selects which bootloader the device is running.
+The possible values are either ``B0`` or ``MCUBOOT``.
+This information about bootloader are checked during both the connection to the device and during the update image compatibility verification.
+If the device does not present information about underlying bootloader or the bootloader type is incompatible, the fwupd tool will fail the image update.
+If the bootloader information contained withing a manifest being part of the update image package differ from the information presented by the device, the update procedure will fail as well.
+
+.. note::
+   Since the ``nordic_hid`` plugin communicate with the device using the :ref:`nrf_desktop_config_channel`, it will not allow device update through the :ref:`nrf_desktop_bootloader_serial_dfu`.
+
+To add a new device a pull request must be opened to the ``fwupd`` repository with a new entry to the :file:`nordic-hid.quirk` file.
+
+LVFS and update image preparation
+---------------------------------
+
+The LVFS project host firmware images that can be downloaded by Linux machines and used by the fwupt tool for firmware update of compatible devices.
```

Comment:
s/The LVFS project host firmware images/The LVFS project hosts firmware images

and

s/the fwupt tool/fwupd